CS483/CS483/Kartaclysm/Components/HUD/ComponentHudAbility.cpp

```diff
@@ -0,0 +1,130 @@
+//----------------------------------------------------------------------------
+// ComponentHudAbility.cpp
+// Author: Bradley Cooper
+//
+// Component that handles an ability icon on a player's HUD.
+//----------------------------------------------------------------------------
+
+#include "ComponentHudAbility.h"
+
+namespace Kartaclysm
+{
+	ComponentHudAbility::ComponentHudAbility(
+		HeatStroke::GameObject* p_pGameObject,
+		const std::string& p_strMTLFileName,
+		const std::string& p_strMaterialName,
+		const std::string& p_strAbility
+		) :
+		HeatStroke::ComponentSprite(p_pGameObject, p_strMTLFileName, p_strMaterialName),
+		m_strEventName("Player0_" + p_strAbility + "_HUD"),
+		m_bReady(false)
+	{
+		m_pDelegate = new std::function<void(const HeatStroke::Event*)>(std::bind(&ComponentHudAbility::AbilityCallback, this, std::placeholders::_1));
+		HeatStroke::EventManager::Instance()->AddListener(m_strEventName, m_pDelegate);
+	}
+
+	ComponentHudAbility::~ComponentHudAbility()
+	{
+		HeatStroke::EventManager::Instance()->RemoveListener(m_strEventName, m_pDelegate);
+		delete m_pDelegate;
+		m_pDelegate = nullptr;
+	}
+
```

Comment:
I added a macro DELETE_IF in Common.h which could be useful to you in the future for deleting things like this.